platform/server/elasticsearch/api.ts

```diff
@@ -23,20 +25,20 @@ export function registerElasticsearchRoutes(
         })
       }
     },
-    async (elasticsearch, req) => {
+    async (req, res) => {
       // WOHO! Both of these are typed!
       log.info(`field param: ${req.params.field}`);
       log.info(`query param: ${req.query.key}`);
 
       log.info('request received on [data] cluster');
 
-      const cluster = await elasticsearch.getClusterOfType('data');
+      const cluster = await service.getScopedCluster('data', req.headers);
 
-      log.info('got [data] cluster, now calling it');
+      log.info('got scoped [data] cluster, now calling it');
 
-      const response = await cluster.withRequest(req, (client, headers) =>
-        client.search({})
-      );
+      // TODO: for some reason my typescript
+      // is saying `search` isn't on this object
+      const response = await cluster.search({});
```

Comment:
There is no `search` method on https://github.com/archanid/kibana/blob/041b644b719beb4f540be9b68cd3fe666a7360c3/platform/server/elasticsearch/Cluster.ts#L6. Before we returned the elasticsearch-js client directly, which has a search.

Reply:
Should we?

Reply:
Add a `search`? In the short-term we can use `callWithRequest`, and then maybe add `search` once we've got that first stuff merged?